luminoth/eval.py

```diff
@@ -376,42 +365,71 @@ def evaluate_once(config, writer, saver, ops, checkpoint,
                 if track_end - track_start > 20.:
                     click.echo(
                         '{} processed in {:.2f}s (global {:.2f} images/s, '
-                        'period {:.2f} images/s, avg dets {:.2f}, '
-                        'max dets {} on {})'.format(
+                        'period {:.2f} images/s)'.format(
                             total_evaluated, track_end - start_time,
                             total_evaluated / (track_end - start_time),
                             track_count / (track_end - track_start),
-                            period_detected_count / track_count,
-                            period_detected_max, period_max_img
                         ))
                     track_count = 0
                     track_start = track_end
-                    period_detected_count = 0
-                    period_detected_max = 0
 
         except tf.errors.OutOfRangeError:
 
             # Save final evaluation stats into summary under the checkpoint's
             # global step.
-            map_at_iou, per_class_at_iou = calculate_map(
-                output_per_batch, config.model.network.num_classes,
-                iou_threshold
+            ap_per_class, ar_per_class = calculate_metrics(
+                output_per_batch, config.model.network.num_classes
             )
 
+            map_at_50 = np.mean(ap_per_class[:, 0])
+            map_at_75 = np.mean(ap_per_class[:, 5])
+            map_at_range = np.mean(ap_per_class)
+            mar_at_range = np.mean(ar_per_class)
+
             tf.logging.info('Finished evaluation at step {}.'.format(
                 checkpoint['global_step']))
             tf.logging.info('Evaluated {} images.'.format(total_evaluated))
-            tf.logging.info('mAP@{}@{} = {:.2f}'.format(
-                iou_threshold, min_probability, map_at_iou))
 
-            # TODO: Find a way to generate these summaries automatically, or
-            # less manually.
+            tf.logging.info(
+                'Average Precision (AP) @ [0.50] = {:.3f}'.format(map_at_50)
+            )
+            tf.logging.info(
+                'Average Precision (AP) @ [0.75] = {:.3f}'.format(map_at_75)
+            )
+            tf.logging.info(
+                'Average Precision (AP) @ [0.50:0.95] = {:.3f}'.format(
+                    map_at_range
+                )
+            )
+            tf.logging.info(
+                'Average Recall (AR) @ [0.50:0.95] = {:.3f}'.format(
+                    mar_at_range
+                )
+            )
+
+            for idx, val in enumerate(ap_per_class[:, 0]):
+                tf.logging.debug(
```

Comment:
Do some of your Python formatting magic for doing something similar to:
<img width="300" alt="image" src="https://user-images.githubusercontent.com/653497/37151029-b5ef4a10-22b2-11e8-9258-0f8077c997d9.png">
?